core/unroller.h

```diff
@@ -43,7 +43,7 @@ class Unroller
   smt::UnorderedTermMap & var_cache_at_time(unsigned int k);
 
   const TransitionSystem & ts_;
-  smt::SmtSolver & solver_;
+  const smt::SmtSolver & solver_;
```

Comment:
shouldn't this be a smart-pointer without ref?

Reply:
Ahh right we said we would not use references for solvers anywhere, right? Changed with 397a95a03d3e580fde5604e398a3c0d21b831dfb